pyiron_atomistics/sphinx/structure.py

```diff
@@ -2,100 +2,99 @@
 # Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
 # Distributed under the terms of "New BSD License", see the LICENSE file.
 
-from collections import OrderedDict
+import re
 import numpy as np
 import scipy.constants
+from pyiron_atomistics.atomistics.structure.parser_base import keyword_tree_parser
 from pyiron_atomistics.atomistics.structure.atoms import Atoms
 from pyiron_atomistics.atomistics.structure.periodic_table import PeriodicTable
 
-__author__ = "Sudarsan Surendralal, Jan Janssen"
+__author__ = "Christoph Freysoldt"
 __copyright__ = (
-    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Copyright 2023, Max-Planck-Institut für Eisenforschung GmbH - "
     "Computational Materials Design (CM) Department"
 )
-__version__ = "1.0"
-__maintainer__ = "Sudarsan Surendralal"
-__email__ = "surendralal@mpie.de"
+__version__ = "2.0"
+__maintainer__ = "Christoph Freysoldt"
+__email__ = "freysoldt@mpie.de"
 __status__ = "production"
-__date__ = "Feb 4, 2018"
+__date__ = "Dec 8, 2023"
 
 BOHR_TO_ANGSTROM = (
     scipy.constants.physical_constants["Bohr radius"][0] / scipy.constants.angstrom
 )
 
+class struct_parser(keyword_tree_parser):
+    """
+    This class reads one or more structures in sx format.
+    """
+    def __init__ (self, file):
+        super().__init__({'structure': self.parse_structure})
+        self.configs = []
+        self.parse (file)
+
+    def parse_structure(self):
+        """ Parses structure{} blocks"""
+        self.keylevels.append ({
+            'cell': self.parse_cell,
+            'species' : self.parse_species})
+        self.extract_via_regex('structure')
+        # --- initialize for next structure
+        self.cell = None
+        self.positions = []
+        self.species = []
+        self.indices = []
+        self.ispecies=-1
```

Comment:
Do these need to be attributes or can they be just local variables?  They are only used to instantiate `Atoms` one L52 and below, or not?